src/mon/OSDMonitor.cc

```diff
@@ -8985,9 +9025,17 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       err = 0;
       goto reply;
     }
-    if (rel == CEPH_RELEASE_LUMINOUS) {
-      if (!HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_LUMINOUS)) {
-	ss << "not all up OSDs have CEPH_FEATURE_SERVER_LUMINOUS feature";
+    assert(osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS);
+    if (rel == CEPH_RELEASE_MIMIC) {
+      if (!osdmap.get_num_up_osds() && sure != "--yes-i-really-mean-it") {
+        ss << "Not advisable to continue since no OSDs are up. Pass "
+           << "--yes-i-really-mean-it if you really wish to continue.";
+        err = -EPERM;
+        goto reply;
+      }
+      if ((!HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_MIMIC))
+           && sure != "--yes-i-really-mean-it") {
+	ss << "not all up OSDs have CEPH_FEATURE_SERVER_MIMIC feature";
```

Comment:
...and here.  It will always be a bad idea for them to do this (via the --force option) on luminous.